tests/unitary/test_reward_unitary.py

```diff
@@ -0,0 +1,141 @@
+def test_notify_without_distributor(multi, reward_token, alice):
+    with brownie.reverts():
+        multi.notifyRewardAmount(reward_token, 10 ** 10, {"from": alice})
+
+
+# Reward per token paid accurate?
+def test_reward_per_token_paid(multi, reward_token, alice, chain):
+    reward_token.approve(multi, 10 ** 19, {"from": alice})
+    multi.setRewardsDistributor(reward_token, alice, {"from": alice})
+    multi.stake(10 ** 10, {"from": alice})
+
+    for i in range(5):
+        _last_val = multi.userRewardPerTokenPaid(alice, reward_token)
+        multi.notifyRewardAmount(reward_token, 10 ** 10, {"from": alice})
+        chain.mine(timedelta=60)
+        multi.getReward()
+        assert multi.userRewardPerTokenPaid(alice, reward_token) > _last_val
+
+
+# Check block.timestamp before periodFinish
+def test_notify_reward_before_period_finish(multi, reward_token, alice, chain):
+    reward_token.approve(multi, 10 ** 19, {"from": alice})
+    multi.setRewardsDistributor(reward_token, alice, {"from": alice})
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+
+    # Initial rate is set
+    initial_rate = 10 ** 15 // 60
+    assert multi.rewardData(reward_token)["rewardRate"] == initial_rate
+
+    # Within duration, update reward
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+    assert multi.rewardData(reward_token)["rewardRate"] > initial_rate
+
+    # Wait for duration to expire
+    chain.mine(timedelta=1000)
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+    assert multi.rewardData(reward_token)["rewardRate"] == initial_rate
+
+    reward_token.approve(multi, 10 ** 19, {"from": alice})
+    multi.setRewardsDistributor(reward_token, alice, {"from": alice})
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+    with brownie.reverts(
+        "Previous rewards period must be complete before changing the duration for the new period"
+    ):
+        multi.setRewardsDuration(reward_token, 9999999, {"from": alice})
+
+
+# Cannot set previous rewards period
+def test_reward_period_finish(multi, reward_token, alice):
+    reward_token.approve(multi, 10 ** 19, {"from": alice})
+    multi.setRewardsDistributor(reward_token, alice, {"from": alice})
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+    with brownie.reverts(
+        "Previous rewards period must be complete before changing the duration for the new period"
+    ):
+        multi.setRewardsDuration(reward_token, 9999999, {"from": alice})
+
+
+# Can set duration after end
+def test_update_reward_duration(multi, reward_token, alice, chain):
+    reward_token.approve(multi, 10 ** 19, {"from": alice})
+    multi.setRewardsDistributor(reward_token, alice, {"from": alice})
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+    chain.mine(timedelta=100)
+    multi.setRewardsDuration(reward_token, 1000, {"from": alice})
```

Comment:
this test could use an assertion at the end, maybe that the reward duration successfully changes to `1000`?